### ISSUE TITLE
In ./create_data.py

### ISSUE THREAD

**Issue by freerailway** [feiyu1990/Skeleton-key#1]
1. In line 373,

>     dset = np.array((N, 256, 256, 3), dtype='uint8')

It produces a 4 element array [N, 256, 256, 3] rather than an array in that shape.
It might be corrected as

>     dset = np.zeros((N, 256, 256, 3), dtype='uint8')

.

2. In lines 374-386, the indents of 2 blanks are not unified with other lines with 4 blanks.

3. In line 405,
    vocab = build_vocab(words_all, params)
It reports an error because _words_all_ has a different structure from _imgs_. Maybe writing a new function _build_vocab_attr_ is a solution.

4. In line 406-408,

>     itow = {i + 1: w for i, w in enumerate(vocab)}  # a 1-indexed vocab translation table
>     json.dump(itow, open(data_root + 'train/ix_to_word_attribute.json', 'w'))
>     json.dump(wtoi, open(data_root + 'train/word2ixs_attr.json', 'w'))

_wtoi_ is the stem dictionary rather than attr. Maybe using _itow_attr_ and _wtoi_attr_ and insert a line

>     wtoi_attr = {w:i+1 for i,w in enumerate(vocab)} # inverse table

is correct.

5. In line 419,

>     out.create_dataset('idx', data=idx_all[:5000], dtype='uint32')

It may raise an error by writing unicode characters in to an h5 file. Same problem in line 558

>     f.create_dataset("labels", dtype='uint32', data=img_idx)

.

6.  In line 431 an 439,

>     out = h5py.File(data_root + 'test_caption_stem.h5', 'w')
>     out = h5py.File(data_root + 'test_idx.h5', 'w')

The directory may be data_root + 'test/'.

**Comment by caoyushe** [comment]
@freerailway Hi, did you run this coce, and trained your own models?

**Comment by freerailway** [comment]
@Jade999   Yes. And I tried 2~3 times to train the models after modifying this file, all failed. I am not very confident that my modification follows the correct intention of the authors, so I am not sure it is a problem of data processing or training strategy.

**Comment by caoyushe** [comment]
@freerailway thank you for your answer, I wonder if  you colud share your modified create_data.py with me, I think we can work together for this code. my email is :  caoyushe@163.com,  thank you very much.

**Comment by caoyushe** [comment]
@freerailway according to your advise, I've modified the create_data.py file,  I can send it to you if you agree and you can compare it with your implementaion.

**Comment by caoyushe** [comment]
@freerailway and in train_1level.py file， line 15: caption_idx = data['first_layer_label2imgid'], but the term "first_layer_label2imgid" never occured when we creat train_caption.h5. I really have not made it. Call for help.

**Comment by mukeshmithrakumar** [comment]
Hei guys, has anyone made this code work? @Jade999 , @feiyu1990 , @freerailway

**Comment by CherishineNi** [comment]
how to deal with coco annotations datasets? what is coco_raw.json? how can I get it? @Jade999 @freerailway @mukeshmithrakumar @feiyu1990
